drivers/android/process.rs

```diff
@@ -855,12 +872,11 @@ impl FileOperations for Process {
         // Drop all references. We do this dance with `swap` to avoid destroying the references
         // while holding the lock.
         let mut refs = obj.node_refs.lock();
-        let mut node_refs = BTreeMap::new();
-        swap(&mut refs.by_handle, &mut node_refs);
+        let mut node_refs = replace(&mut refs.by_handle, RBTree::new());
         drop(refs);
 
         // Remove all death notifications from the nodes (that belong to a different process).
-        for info in node_refs.values_mut() {
+        for (_, info) in &mut node_refs {
```

Comment:
Probably better to add `values`, `values_mut` to RBTree instead?

Reply:
Probably better, yes. But not a priority at the moment as the code seems readable enough.

Reply:
It should be a straightforward addition if `impl` syntax is used.

```rust
pub fn keys(&self) -> impl Iterator<Item = &'_ K> {
    self.iter().map(|(k, _)| k)
}

pub fn values(&self) -> impl Iterator<Item = &'_ V> {
    self.iter().map(|(_, v)| v)
}

pub fn values_mut(&mut self) -> impl Iterator<Item = &'_ mut V> {
    self.iter_mut().map(|(_, v)| v)
}
```

Reply:
Oh, nice. I didn't know about this syntax, I thought I had to define a new type. I was dreading that and writing more documentation.

This should be straightforward enough. I'll update this PR and the rbtree one. Thanks!